### PR TITLE
Add shopify api shop tests

### PR DESCRIPTION
**Pull request by AntonoffEvgeniy**

### CODE REVIEW

File: ShopifyTests/API/Mocks/GraphClientMock.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+@testable import MobileBuySDK
+@testable import Shopify
+
+class GraphClientMock: Graph.Client {
+    var returnedResponse: Storefront.QueryRoot?
+    var returnedError: Graph.QueryError?
+    
+    override func queryGraphWith(_ query: Storefront.QueryRootQuery, cachePolicy: Graph.CachePolicy?, retryHandler: Graph.RetryHandler<Storefront.QueryRoot>?, completionHandler: @escaping Graph.QueryCompletion) -> Task {
+
```
Comment (radyslavkrechet) [inline ShopifyTests/API/Mocks/GraphClientMock.swift]: Remove single line, please.

File: ShopifyTests/API/Mocks/GraphClientMock.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        completionHandler(returnedResponse, returnedError)
+        
+        let client = Graph.Client(shopDomain: "shopDomain", apiKey: "apiKey")
+        let queryGraph = Storefront.buildQuery { $0
+            .shop { $0
+                .name()
+            }
+        }
+            
+        let task = client.queryGraphWith(queryGraph) { (response, error) in }
```
Comment (radyslavkrechet) [inline ShopifyTests/API/Mocks/GraphClientMock.swift]: Can we not create client object and use `super`? Also can we not create `task` variable and return value at once?

**Review by radyslavkrechet — CHANGES_REQUESTED** [review]

File: ShopifyTests/API/Mocks/GraphClientMock.swift (hunk 2 above)
Comment (plzen) [inline ShopifyTests/API/Mocks/GraphClientMock.swift]: Are you sure that this doesn't send live request to Shopify?

File: ShopifyTests/API/ShopifyAPI/ShopifyAPIShopSpec.swift [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        }
+        
+        describe("when get shop info called") {
+            context("if success") {
+                beforeEach {
+                    clientMock.returnedResponse = try! Storefront.QueryRoot(fields: ["shop": ShopifyAPITestHelper.shop])
+                }
+                
+                it("should return success") {
+                    shopifyAPI.getShopInfo(callback: { (shop, error) in
```
Comment (radyslavkrechet) [inline ShopifyTests/API/ShopifyAPI/ShopifyAPIShopSpec.swift]: Don't use `callback` keyword.

File: ShopifyTests/API/Mocks/GraphClientMock.swift (hunk 2 above)
Comment (radyslavkrechet) [reply ShopifyTests/API/Mocks/GraphClientMock.swift]: I think that the best solution is next:

```
override func queryGraphWith(_ query: Storefront.QueryRootQuery, cachePolicy: Graph.CachePolicy?, retryHandler: Graph.RetryHandler<Storefront.QueryRoot>?, completionHandler: @escaping Graph.QueryCompletion) -> Task {
    completionHandler(returnedResponse, returnedError)
        
    return URLSessionDataTask()
}
```

**Review by radyslavkrechet — APPROVED** [review]

**Review by plzen — APPROVED** [review]

**Pull request merged**